app/src/main/java/com/infomaniak/mail/ui/newMessage/selectMailbox/SelectMailboxScreen.kt

```diff
@@ -66,132 +73,200 @@ fun SelectMailboxScreen(
-                            modifier = Modifier.padding(Margin.Medium),
-                            selectedMailbox = it
-                        )
-                    }
-                } else {
-                    Column {
-                        LazyColumn(
-                            modifier = Modifier.padding(Margin.Medium),
-                            verticalArrangement = Arrangement.spacedBy(Margin.Mini),
-                        ) {
-                            items(usersWithMailboxes) { userWithMailboxes ->
-                                AccountMailboxesDropdown(
-                                    userWithMailboxes = userWithMailboxes,
-                                    onClickMailbox = { mailbox ->
-                                        onMailboxSelected(mailbox)
-                                    }
-                                )
-                            }
-                        }
-                        Spacer(modifier = Modifier.weight(1f))
-                        selectedMailbox?.let {
-                            SelectedMailboxIndicator(
-                                modifier = Modifier.padding(Margin.Medium),
-                                selectedMailbox = it
-                            )
-                        }
-                    }
-                }
+                SelectedMailboxBottom(selectedMailbox)
             }
         },
-        topButton = {
-            LargeButton(
-                modifier = it.padding(horizontal = Margin.Medium),
-                title = stringResource(R.string.buttonContinue),
-                enabled = { selectedMailbox != null }
-            ) {
-                selectedMailbox?.let { onContinue(selectedMailbox) }
-            }
+        topButton = { modifier ->
+            TopButton(modifier, uiState, onContinueWithMailbox)
         },
-        bottomButton = bottomButton.takeIf{ !selectingAnotherUser.value }
+        bottomButton = { modifier ->
+            BottomButton(modifier, uiState, onMailboxSelected, onChooseAnotherMailbox)
+        }
+    )
+}
+
+@Composable
+private fun Header() {
+    Image(
+        imageVector = ImageVector.vectorResource(R.drawable.illustration_mailbox_ellipsis_bubble),
+        contentDescription = null
+    )
+    Text(
+        style = Typography.h2,
+        maxLines = 1,
+        text = stringResource(R.string.composeMailboxCurrentTitle)
     )
 }
 
 @Composable
-@Preview(name = "(1) Light")
-@Preview(name = "(2) Dark", uiMode = Configuration.UI_MODE_NIGHT_YES or Configuration.UI_MODE_TYPE_NORMAL)
+private fun ColumnScope.ScrollableContent(
+    uiState: () -> UiState,
+    usersWithMailboxes: List<UserMailboxesUi>,
+    onMailboxSelected: (SelectedMailboxUi?) -> Unit
+) {
+    uiState().let { uiState ->
+        LazyColumn(
+            modifier = Modifier
+                .padding(horizontal = Margin.Medium)
+                .fillMaxWidth()
+                .weight(1f),
```

Comment:
It fills easier to follow around and more appropriate to have these layout procurations be defined at the callplace by providing a Modifier instance. This way the column is the only one that manages the layout aspect and the `ScrollableContent` and ScrollableContent's responsability can be reduced